.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: '^(prover_snapshots)|(generated_definitions)|(c_emulator/SoftFloat-3e)'
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+        # Do not strip trailing whitespace from patches or SVG images.
+        # TODO: the objdump outputs should probably not be part of the repo.
+        exclude: '.*\.(diff|dump|patch|svg)'
+    -   id: end-of-file-fixer
+        # Do not strip trailing newlines from patches or SVG images.
+        # TODO: the objdump outputs should probably not be part of the repo.
+        exclude: '.*\.(diff|dump|patch|svg)'
+    -   id: check-yaml
+    -   id: check-added-large-files
```

Comment:
According to https://pre-commit.com/hooks.html, we can also add a clang-format hook ? Maybe a good idea since #263

Reply:
Yes I have a local change to add that and will to push it once these two PRs have been merged.